dts/bindings/base/mem-region.yaml

```diff
@@ -1,6 +1,9 @@
 # Copyright (c) 2021, CSIRO
 # SPDX-License-Identifier: Apache-2.0
 # Common fields for devices resulting in linker memory regions
+#
+# NOTE: Every compatible importing this YAML binding should be present in the
+# list in include/linker/devicetree_regions_mpu.h
```

Comment:
I understand the reasons behind this, but it's a bit ugly. Have you thought about other options? Maybe extra dt tooling? using an extra compatible (not sure if feasible)?

Reply:
Using an extra compatible is technically feasible because [DT_FOREACH_STATUS_OKAY](https://docs.zephyrproject.org/latest/reference/devicetree/api.html#c.DT_FOREACH_STATUS_OKAY) works even for compatibles that do not have matching bindings.

For example, there is no binding for compatible `abcd` in this example:

```
$ cat samples/hello_world/app.overlay
/ {
	foo {
		compatible = "gpio-keys", "abcd";
	};

	bar {
		compatible = "pwm-leds", "abcd";
	};
};

$ cat samples/hello_world/src/main.c
/*
 * Copyright (c) 2012-2014 Wind River Systems, Inc.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

#include <zephyr.h>
#include <sys/printk.h>

#define PATH_COMMA(node_id) DT_NODE_PATH(node_id),

void main(void)
{
	const char *paths[] = {
		DT_FOREACH_STATUS_OKAY(abcd, PATH_COMMA)
	};
	for (int i = 0; i < ARRAY_SIZE(paths); i++) {
		printk("%s\n", paths[i]);
	}
}
```

But it still works:

```
$ west build -b native_posix -t run samples/hello_world
[...]
*** Booting Zephyr OS build v2.7.99-2322-gb326aabf73ce  ***
/foo
/bar
```

I'm not saying this is the right way to go, just saying no extra tooling is needed if you want to just use an extra compatible, regardless of whether you define a matching binding for it.

If you do write a matching binding, then the specific binding should come first, and the generic binding second, as usual (https://docs.zephyrproject.org/latest/guides/dts/bindings.html#other-ways-nodes-are-matched-to-bindings).

Reply:
So, what's the preference here? If I have to rework this I'd like to have a precise way forward.

Reply:
Any solution that avoids this:

```c
/*
 * List of compatibles whose YAML files are including mem-region.yaml to
 * support the 'zephyr,memory-region-mpu' property.
 */
#define _DT_COMPATIBLES mmio_sram, arm_dtcm, arm_itcm, nxp_imx_dtcm, nxp_imx_itcm,	\
			st_stm32_ccm, st_stm32_backup_sram
```

would be welcome, if technically possible.

Reply:
alright, going for the additional compatible